pygments/lexers/elpi.py

```diff
@@ -152,21 +152,24 @@ class ElpiLexer(RegexLexer):
             (r'"', String.Double, '#pop'),
         ],
         'elpi-quote': [
-            (r'\{\{', Punctuation, '#push'),
             (r'\}\}', Punctuation, '#pop'),
-            (r"(lp:)((?=[A-Z_]){})".format(constant_re), bygroups(Keyword, Name.Variable)),
-            (r"[^l\}]+", Text),
-            (r"l|\}", Text),
+            (r"\s", Text.Whitespace),
+            (r"(lp:)({})".format("{{"), bygroups(Number, Punctuation), 'elpi-quote-exit'),
+            (r"(lp:)((?=[A-Z_]){})".format(constant_re), bygroups(Number, Name.Variable)),
+            (r".", Text),
```

Comment:
Also `.+`, otherwise you get one match per character (see your output - you want multiple consecutive characters to match?

Reply:
this is a good idea! Nonetheless, .+ is overly aggressive, as I don't intend to capture all characters up to the next \n. 
I can opt for a more nuanced approach, drawing inspiration from .+ but employing negative lookahead.